tests/testSprocs.js

```diff
@@ -0,0 +1,304 @@
+var ERR = require('async-stacktrace');
+var assert = require('chai').assert;
+
+var sqldb = require('@prairielearn/prairielib').sqldb;
+var sqlLoader = require('@prairielearn/prairielib').sqlLoader;
+var sql = sqlLoader.loadSqlEquiv(__filename);
+var helperDb = require('./helperDb');
+
+
+var caa_reservation_tests = function(assessment_id, exam_id, second_assessment_id, expectWideOpen=false, seeOtherExams=false) {
+
+    var expectedWord = 'fail';
+    var expectedBool = false;
+
+    // Handle the special case without any linking
+    if (expectWideOpen) {
+        expectedWord = 'pass';
+        expectedBool = true;
+    }
+
+    it(`${expectedWord} for student inside start_date/end_date, no reservation, assessment ${assessment_id}`, function(callback) {
+        var params = [
+            assessment_id,
+            'Exam',
+            'Student',
+            1002,
+            'instructor@school.edu',
+            '2010-07-07 06:06:06-00',
+            'US/Central',
+        ];
+
+        sqldb.call(`check_assessment_access`, params, (err, result) => {
+            if (ERR(err, result)) return;
+            assert.strictEqual(result.rows[0].authorized, expectedBool);
+            callback();
+        });
+    });
+
+    describe(`with checked-in reservation for student for exam ${exam_id}`, () => {
+
+        before(`create checked-in reservation for student for exam ${exam_id}`, function(callback) {
+            sqldb.query(sql.insert_ps_reservation, {exam_id}, (err, _result) => {
+                if (ERR(err,callback)) return;
+                callback();
```

Comment:
By convention we normally write this as `callback(null)`, which is technically slightly different. I guess mocha doesn't care, but I'd rather stick with the convention anyway.